pkg/operator/ceph/object/controller.go

```diff
@@ -192,36 +192,6 @@ func (r *ReconcileCephObjectStore) reconcile(request reconcile.Request) (reconci
 		updateStatus(r.opManagerContext, k8sutil.ObservedGenerationNotAvailable, r.client, request.NamespacedName, cephv1.ConditionProgressing, buildStatusInfo(cephObjectStore))
 	}
 
-	// Make sure a CephCluster is present otherwise do nothing
-	cephCluster, isReadyToReconcile, cephClusterExists, reconcileResponse := opcontroller.IsReadyToReconcile(r.opManagerContext, r.client, request.NamespacedName, controllerName)
-	if !isReadyToReconcile {
-		// This handles the case where the Ceph Cluster is gone and we want to delete that CR
-		// We skip the deleteStore() function since everything is gone already
-		//
-		// Also, only remove the finalizer if the CephCluster is gone
-		// If not, we should wait for it to be ready
-		// This handles the case where the operator is not ready to accept Ceph command but the cluster exists
-		if !cephObjectStore.GetDeletionTimestamp().IsZero() && !cephClusterExists {
-			// Remove finalizer
-			err := opcontroller.RemoveFinalizer(r.opManagerContext, r.client, cephObjectStore)
-			if err != nil {
-				return reconcile.Result{}, *cephObjectStore, errors.Wrap(err, "failed to remove finalizer")
-			}
-
-			// Return and do not requeue. Successful deletion.
-			return reconcile.Result{}, *cephObjectStore, nil
-		}
-
-		return reconcileResponse, *cephObjectStore, nil
-	}
-	r.clusterSpec = &cephCluster.Spec
-
-	// Populate clusterInfo during each reconcile
-	r.clusterInfo, _, _, err = opcontroller.LoadClusterInfo(r.context, r.opManagerContext, request.NamespacedName.Namespace, r.clusterSpec)
-	if err != nil {
-		return reconcile.Result{}, *cephObjectStore, errors.Wrap(err, "failed to populate cluster info")
-	}
-
```

Comment:
I'm concerned about stripping out this code wholesale. It is still important, especially the part that "handles the case where the Ceph Cluster is gone and we want to delete that CR"